src/ostorlab/runtimes/cloud/runtime.py

```diff
@@ -46,19 +52,83 @@ def can_run(self, agent_group_definition: definitions.AgentGroupDefinition) -> b
         Returns:
             True if can run, false otherwise.
         """
-        pass
+        try:
```

Comment:
It would be nice to group individuals checks to make the check intention more explicit.